awx/main/models/inventory.py

```diff
@@ -1419,6 +1158,20 @@ def clean_source_path(self):
             raise ValidationError(_("Cannot set source_path if not SCM type."))
         return self.source_path
 
+    def clean_source_vars(self):
+        injector = self.injectors.get(self.source)
+        source_vars = dict(self.source_vars_dict) # make a copy
+        if injector and self.source_vars_dict.get('plugin', '') != injector.get_proper_name():
+            source_vars['plugin'] = injector.get_proper_name()
+        elif not injector:
```

Comment:
`scm` source type requires the user to specify `plugin: a.b.c`, we can't do it for them.